### PR TITLE
DPC-3740: Fix assets pipeline in deployed environments

### PR DESCRIPTION
**Pull request by kyeah**
## 🎫 Ticket

https://jira.cms.gov/browse/DPC-3740

## 🛠 Changes

- Link USWDS assets within the Sprockets manifest.js file to ensure they are marked for production usage
- Update javascript_include_tag to match new asset paths
- Ensure assets are shipped and referenced under the `/portal` subpath, so they are accessible in deployed environments.
- Remove live-reload dependencies since they are not usable with the new subpath implementation
- Add lots of documentation
- Replace global volume mount in docker-compose to ensure that node_modules and precompiled assets are not overwritten in local development.

## ℹ️ Context for reviewers

In the dev environment, we are currently running into two clear issues related to the assets pipeline:

1. When visiting the lookbook pages, Lookbook stylesheets are not being loaded correctly:

|Problem Image|Fix|
|---|---|
|CleanShot 2023-12-03 at 14 55 13@2x|The assets are served under `dev.dpc.cms.gov/lookbook-assets`, but they need to be served under `dev.dpc.cms.gov/portal/lookbook-assets` to avoid being routed to the dpc-web application.|

2. When visiting the user sign in page, we are receiving 500 error messages with this log in splunk:

```
"ActionView::Template::Error","Asset `@uswds/uswds/dist/js/uswds-init.min.js` was not declared to be precompiled in production.\nDeclare links to your assets in `app/assets/config/manifest.js`.\n\n  //= link @uswds/uswds/dist/js/uswds-init.min.js\n\nand restart your server"
```

In deployed environments, any usable assets need to be declared for production usage in the `manifest.js` file. We also need to ensure that these assets are served under the `/portal` prefix.

This PR ensures that assets are served under the `/portal` prefix by updating the `config.ru` file to map all routes, including assets, to that prefix. Previously we were using the `scope "portal"` method, which works for normal routes but not for assets.

### Live Reload

Since we are no longer using `scope "portal"`, this breaks the live-reload functionality from `rails_live_reload` and `lookbook`. I played around with it but couldn't figure out a solution to this, so during development, we will need to refresh the web page to see updated changes.

### Local Development
I confirmed that both the localhost:3100/portal/users/sign_in and localhost:3100/portal/lookbook pages were accurate when running the portal locally.

### Production Builds
I tested the production builds by replicating it locally:

1. Comment out volume mounts in docker-compose.portals.yml
2. Add the following env vars in docker-compose file: RAILS_ENV=production, ENV=dev, SECRET_KEY_BASE=dummy
3. Run `make portals` and `make start-portals`. 
4. Verify that both localhost:3100/portal/users/sign_in and localhost:3100/portal/lookbook still work properly.

## 🔒 Security Implications

- [ ] This PR adds a new software dependency or dependencies.
- [ ] This PR modifies or invalidates one or more of our security controls.
- [ ] This PR stores or transmits data that was not stored or transmitted before.
- [ ] This PR requires additional review of its security implications for other reasons.

### CODE REVIEW

**Review by jdettmannnava — APPROVED** [review]
LGTM

File: dpc-portal/config/environments/development.rb [hunk 1]
```diff
@@ -63,6 +64,10 @@
   # Suppress logger output for asset requests.
   config.assets.quiet = true
 
+  # Fall back to assets pipeline at runtime if a precompiled asset is mising.
```
Comment (MEspositoE14s) [inline dpc-portal/config/environments/development.rb:67]: Proof that I really do read through all of these: "mising" -> "missing"

**Review by MEspositoE14s — APPROVED** [review]
Everything works locally, and LookBook seems super cool.

File: dpc-portal/config/application.rb [hunk 2]
```diff
@@ -13,6 +13,10 @@ class Application < Rails::Application
     # Initialize configuration defaults for originally generated Rails version.
     config.load_defaults 7.0
 
+    # Set the relative_url_root at runtime, which will be used in various places
+    # to ensure that we are serving everything under the portal scope.
+    config.relative_url_root = '/portal'
```
[the remark this answers is not in the document]
Comment (kyeah) [reply dpc-portal/config/application.rb:18]: Fixed this in https://github.com/CMSgov/dpc-app/pull/1988/commits/0104a19e2ef30ebe0534ca4b2936049bac5a3d55! Verified locally and in dev environment that `/portal/health_check` routes properly.

**Pull request merged**